openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py

```diff
@@ -273,16 +274,6 @@ def add_move_id_field_account_bank_statement_line(env):
 
 
 def add_move_id_field_account_payment(env):
-    openupgrade.logged_query(
-        env.cr,
-        """
-        UPDATE account_payment ap
-        SET currency_id = COALESCE(aj.currency_id, rc.currency_id)
```

Comment:
Don't. account.payment has a computed stored field: https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_payment.py#L396

Reply:
No, `currency_id` is inherited from `account.move`, I updated it at line 346

Reply:
In essence, it inherits from `account.move`, so it only needs to be updated at line 346

Reply:
In line 346, you are updating the `am.currency_id`, nothing to do with `ap.currency_id`, which this query handles. From a database point of view, `am.currency_id` and `ap.currency_id` are two different columns in two different tables. This query you are deleting assures the value of `ap.currency_id` is the same as the value of the compute I linked before.